dissect/target/target.py

```diff
@@ -419,6 +419,8 @@ def _init_os(self) -> None:
 
             if isinstance(os_plugin, plugin.OSPlugin):
                 self._os_plugin = os_plugin.__class__
+            elif issubclass(os_plugin, plugin.OSPlugin):
+                os_plugin = os_plugin.create(self, os_plugin.detect(self))
```

Comment:
I want to set the `_os_plugin` to `WindowsPlugin` in the `SmbLoader` to prevent unnecessary requests to file paths of other operating systems. However, I do need to run the `.detect()` function on the `OSPlugin` in order to find the `sysvol`. Either I do that in the loader itself (in which case you'd also have to do it in future loaders where you want to do this), or I make it a part of the feature of prematurely setting `_os_plugin` that already exists, but currently only really works if you set it to an instance of an `OSPlugin`, instead of a class.